src/ApiRequestPanel.js

```diff
@@ -490,6 +490,25 @@ export class ApiRequestPanel extends AmfHelperMixin(EventsTargetMixin(HeadersPar
     this._computeServerSelectorHidden();
   }
 
+  get baseUri() {
+      if (this._baseUri) {
+        return this._baseUri;
+      }
+      if (this.selectedServerType !== 'server') {
+        return this.selectedServerValue;
+      }
+      return this._baseUri;
+  }
+
+  set baseUri(value) {
```

Comment:
I don't like this idea of `baseUri` returning different values depending on the conditions above. Please, define new getter (like `get effectiveBaseUri() {}` or similar) that only returns the value that should be propagated downwards.